docs/how-to-guides/performing-cluster-maintenance.md

```diff
@@ -0,0 +1,239 @@
+
+```text
+Model      Controller  Cloud/Region         Version  SLA          Timestamp
+microceph  overlord    localhost/localhost  3.6.3    unsupported  14:09:58+08:00
+
+App        Version  Status  Scale  Charm      Channel      Rev  Exposed  Message
+microceph           active      4  microceph  latest/edge  103  no
+
+Unit          Workload  Agent  Machine  Public address  Ports  Message
+microceph/0   active    idle   0        10.42.75.217
+microceph/1*  active    idle   1        10.42.75.46
+microceph/2   active    idle   2        10.42.75.23
+microceph/3   active    idle   3        10.42.75.189
+
+Machine  State    Address       Inst id        Base          AZ  Message
+0        started  10.42.75.217  juju-681fff-0  ubuntu@24.04      Running
+1        started  10.42.75.46   juju-681fff-1  ubuntu@24.04      Running
+2        started  10.42.75.23   juju-681fff-2  ubuntu@24.04      Running
+3        started  10.42.75.189  juju-681fff-3  ubuntu@24.04      Running
+
+MicroCeph deployment summary:
+- juju-681fff-0 (10.42.75.217)
+  Services: mds, mgr, mon, osd
+  Disks: 1
+- juju-681fff-1 (10.42.75.46)
+  Services: mds, mgr, mon, osd
+  Disks: 1
+- juju-681fff-2 (10.42.75.23)
+  Services: mds, mgr, mon, osd
+  Disks: 1
+- juju-681fff-3 (10.42.75.189)
+  Services: mon, osd
+  Disks: 1
+```
+
+## Procedures
+
+### 0. Review the action plan of maintenance mode
+
+The action plan for entering or exiting the maintenance mode can reviewed by using the run dry-run option.
+
+```shell
+juju run microceph/leader exit-maintenance dry-run=True
+juju run microceph/leader enter-maintenance dry-run=True
+```
+
+Some steps in the action plan can be optionally added or removed from the action plan. To see what are the optional
+steps, run
+
+```shell
+juju show-action microceph exit-maintenance
+juju show-action microceph enter-maintenance
+```
+
+### 1. Enter maintenance mode
+
+To put unit `microceph/3` into maintenance mode, and optionally disable the OSD service on that node, run
+
+```shell
+juju run microceph/3 enter-maintenance stop-osds=True
+```
+
+Sample output is:
+
+```text
+Running operation 15 with 1 task
+  - task 16 on unit-microceph-3
+
+Waiting for task 16...
+actions:
+  step-1:
+    description: Check if osds.[4] in node 'juju-681fff-3' are ok-to-stop.
+    error: ""
+    id: check-osd-ok-to-stop-ops
+  step-2:
+    description: Check if there are at least 3 mon, 1 mds, and 1 mgr services in the
+      cluster besides those in node 'juju-681fff-3'
+    error: ""
+    id: check-non-osd-svc-enough-ops
+  step-3:
+    description: Run `ceph osd set noout`.
+    error: ""
+    id: set-noout-ops
+  step-4:
+    description: Assert osd has 'noout' flag set.
+    error: ""
+    id: assert-noout-flag-set-ops
+  step-5:
+    description: Stop osd service in node 'juju-681fff-3'.
+    error: ""
+    id: stop-osd-ops
+errors: ""
+status: success
+```
+
+After entering maintenance mode, the status of the cluster is:
```

Comment:
[* reword *] : " After entering maintenance mode, this is the status of the cluster:"